Modules/socketmodule.c

```diff
@@ -3263,11 +3274,20 @@ sock_getsockopt(PySocketSockObject *s, PyObject *args)
         return NULL;
         }
 #endif
+
+#ifdef AF_RDS
+    if (s->sock_family == AF_RDS) {
+        goto get_buf;
+    }
+#endif
+
     if (buflen <= 0 || buflen > 1024) {
         PyErr_SetString(PyExc_OSError,
                         "getsockopt buflen out of range");
         return NULL;
     }
+
+get_buf:
```

Comment:
It might be a good idea to either a. not use `goto`, or b. only use it under an `#ifdef AF_RDS`. Otherwise, build bots and editors will complain about `get_buf` being unused if `AF_RDS` is not defined.